modules/csg/csg_shape.cpp

```diff
@@ -896,6 +903,17 @@ Ref<Mesh> CSGMesh::get_mesh() {
 	return mesh;
 }
 
+void CSGMesh::set_material_override(const Ref<Material> &p_material) {
```

Comment:
There is actually an existing material override property (and setter/getter) at GeometryInstance level, so I think you can simply just use it.
just call get_material_override()

Reply:
@reduz It doesn't look like CSGMesh inherits from GeometryInstance, but VisualInstance instead which does not have the material_override property.

Reply:
@reduz 
If making CSGShape inherit from GeometryInstance is desired, I can do that and then remove all the material parameters on all the primitives.

Reply:
@reduz, I'm looking for some confirmation either to change the inheritance of the CSGShape, or to leave this alone.

Reply:
Sorry, yes it definitely should inherit from GeometryInstance, this was my mistake.

Reply:
@reduz thanks! I'll get to work on that